core/cpu.cpp

```diff
@@ -618,9 +595,352 @@ u8 CPU::exec(){
+            pc++;
+
+            if(readCarryFlag()){
+                pc = nn_nn;
+                return 16;
+            }
+            else{
+                return 12;
+            }
+        }
+        case 0xDC: {
+            //CALL C, a16
+            //check if carry flag is set and jump if so
+
+            //PC NEEDS TO BE INCREMENTED TWICE ON 16 BIT READ
+            u16 nn_nn = mmu.read16Bit(pc++);
+            pc++;
+
+            if(readCarryFlag()){
+                pushToStack(pc);
+                pc = nn_nn;
+                return 24;
+            }
+            else{
+                return 12;
+            }
+        }
+        case 0xDE: {
+            //SBC A, d8
+            //A=A-n-cy
+            u8 value = mmu.read(pc++);
+            u8 a = getHighByte(af);
+            u8 result = op_sbc(a, value);
+            setHighByte(&af, result);
+            return 8;
+        }
+        case 0xE0: {
+            //LDH (a8), A aka LD ($FF00+a8), A
+            u8 input = mmu.read(pc++);
+            mmu.write(0xFF00 + input, getHighByte(af));
+            return 12;
+        }
+        case 0xE6: {
+            //AND d8
+            u8 n = mmu.read(pc++);
+            u8 a = getHighByte(af);
+            setHighByte(&af, op_and(a, n));
+            return 8;
+        }
+        case 0xE8: {
+            //ADD SP, r8
+            //SP = SP +/- dd ; dd is 8-bit signed number
+            s8 num = mmu.read(pc++);
+            pc = op_add(sp, num);
+            return 16;
+        }
+        case 0xE9: {
+            //JP HL
+            //jump to HL, PC=HL
+            pc = hl;
+            return 4;
+        }
+        case 0xEA: {
+            //LD (a16), A
+            u16 addressToWrite = mmu.read(pc++);
+            pc++;
+
+            mmu.write(addressToWrite, getHighByte(af));
+            return 16;
+        }
+        case 0xEE: {
+            //XOR d8
+            //A=A xor n
+            u8 n = mmu.read(pc++);
+            setHighByte(&af, op_or(getHighByte(af), n));
+            return 8;
+        }
+        case 0xF0: {
+            //LDH A, (a8) aka LD A, ($FF00+a8)
+            u8 input = mmu.read(pc++);
+            setHighByte(&af, (mmu.read(0xFF00+input)));
+            return 12;
+        }
+        case 0xF2: {
+            //ld A,(FF00+C)
+            u8 c = getLowByte(bc);
+            setHighByte(&af, (mmu.read(0xFF00+c)));
+            return 8;
+        }
+        case 0xF3: {
+            //DI
+            //disable interrupts, IME=0
+            /*******************NOT SURE HOW TO DO THIS*******************/
+            ime = false;
+            return 4;
+        }
+        case 0xF6: {
+            //OR d8
+            u8 valueToOr = mmu.read(pc++);
+
+            u8 a = getHighByte(af);
+            a = op_or(a, valueToOr);
+            setHighByte(&af, a);
+            return 8;
+        }
+        case 0xF8: {
+            //LD HL, SP + r8
+            u8 valueToAdd = mmu.read(pc++);
```

Comment:
I think `r8` means signed, so this line should be `s8 valueToAdd ...`

Reply:
Changed